ultra/ultra/adapters/default_observation_image_adapter.py

```diff
@@ -0,0 +1,60 @@
+# MIT License
+#
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import gym
+import numpy as np
+
+from smarts.core.agent_interface import RGB
+from smarts.core.sensors import Observation
+
+
+_WIDTH = 64
+_HEIGHT = 64
+_RESOLUTION = 50 / 64
+
+
+# The space of the adapted observation.
+gym_space: gym.Space = gym.spaces.Box(
+    low=0.0, high=1.0, shape=(_HEIGHT, _WIDTH), dtype=np.float32
+)
+# This adapter requires SMARTS to pass the top-down RGB image in the agent's
+# observation.
+required_interface = {"rgb": RGB(width=_WIDTH, height=_HEIGHT, resolution=_RESOLUTION)}
+
+
+def adapt(observation: Observation) -> np.ndarray:
+    """Adapts a raw environment observation into a numpy.ndarray.
+
+    The raw observation from the environment must include the top down RGB image. See
+    smarts.core.sensors for more information on the Observation type.
+
+    Args:
+        observation (Observation): The raw environment observation received from SMARTS.
+
+    Returns:
+        np.ndarray: A numpy.ndarray of size (64, 64) that is the gray-scale image of the
+            top down RGB image. The gray-scale value for each pixel is calculated as
+            0.2125 * r + 0.7154 * g + 0.0721 * b.
+    """
+    rgb_image = observation.top_down_rgb.data
+    gray_image = np.dot(rgb_image, (0.2125, 0.7154, 0.0721))
+    normalized_gray_image = np.divide(gray_image, 255.0)
+    return normalized_gray_image.astype(np.float32)
```

Comment:
Yeah good question, I hadn't really looked into it much further. However, I did submit Issue #768 to get some help with this potential problem. I can bring the issue up in the SMARTS meeting tomorrow.

Reply:
(Sorry I just realized I hadn't responded to this one)

The position and rotation of the ego vehicle's rectangle never changes in the image, perhaps this will also help the ego vehicle despite the gray-scale values being similar for the ego vehicle and the road. However, I have also taken a look at different weightings that darken the ego vehicle (decrease the red channel weight):
| (0.2125, 0.7154, 0.0721) | (0.15, 0.8, 0.05) | (0.1, 0.8, 0.1) |
|---------------------------|-----------------|----------------|
| <img src="https://user-images.githubusercontent.com/49890535/120840858-fd5c1980-c527-11eb-8249-9a564c23cb8b.png" width="200" height="200" /> | <img src="https://user-images.githubusercontent.com/49890535/120841166-6774be80-c528-11eb-9335-ca4951a330a8.png" width="200" height="200" /> | <img src="https://user-images.githubusercontent.com/49890535/120841316-9723c680-c528-11eb-8c54-3e83cd915f85.png" width="200" height="200" /> |

Reply:
The one on the right looks good! This probably wont make a big difference from the CNN's pov but it will help us if we ever want to see what the policy/NN is seeing.